### ISSUE TITLE
Suggestion: File Dialog should set default extension if entered filename has none.

### ISSUE THREAD

**Issue by Chaircrusher** [VCVRack/Rack#670]
OS (if applicable): All
Version (or "dev" if compiling from source): All

The VCV file dialogs are happy to save a file under any filename extension, meaning that I've accidentally saved VCV patches as 'something.wav' overwriting audio recordings. Also if there's no extension, an appropriate one should be appended.

This conforms to the normal behavior of application.

**Comment by AndrewBelt** [comment]
I think what you're trying to say is 1) that the dialog boxes should filter by extensions on all three OSes for patch loading/saving, and 2) this code https://github.com/VCVRack/Rack/blob/v0.5/src/app/RackWidget.cpp#L74-L79 is not working for some reason.

(1) would require a modification to https://github.com/AndrewBelt/osdialog which is on its TODO list.
Could you investigate why (2) is not working for you?

**Comment by Chaircrusher** [comment]
The business with tacking on ".vcv" is working.  What isn't working is that if there IS an extension, it will use that even if it's a bad idea, like ".wav"

And that interacts with things like the predictive selection when you start typing. You probably leave that up to the OS, and it wouldn't surprise me if the Linux dialog you're using is dumb.

And yeah, it sounds like you're on top of it, so... carry on! ;-)

**Comment by AndrewBelt** [comment]
I think that should be expected behavior. If you want an extension different than .vcv (like .json or .txt), it should respect that and not append .vcv. When overwriting a WAV file, is the OS dialog not warning you for overwriting the file?

**Comment by AScustomWorks** [comment]
Not sure if this belongs here, but how about adding a .vcv file restriction to the "open" dialog?, right now it lets you select any file (and it tries to open it even if not a .vcv file), thanks Andrew! ;)

**Comment by AndrewBelt** [comment]
Yeah, I should probably restrict it. Nobody uses the .json extension anyway.

**Comment by squinkylabs** [comment]
Letting you enter arbitrary extensions seems correct. What doesn't seem correct (to me) is that when I do "file open" on windows, it does not filter on .vcv, it shows me all extensions (*.*)

**Comment by squinkylabs** [comment]
@AndrewBelt : my observation, above, was meant more or less as a bug report. I believe that one windows the file dialogs should start with a *.vcv filter. I thought they were coming up with *.*, which seems like a bug.
